packages/fxa-admin-panel/src/components/PageRateLimiting/index.tsx

```diff
@@ -0,0 +1,270 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React, { useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { useLazyQuery, useMutation, ApolloError } from '@apollo/client';
+import iconSearch from '../../images/icon-search.svg';
+import ErrorAlert from '../ErrorAlert';
+import { BlockStatus } from './BlockStatus';
+import { GET_RATE_LIMITS, UNBLOCK, UNBAN } from './index.gql';
+import { BlockStatus as BlockStatusData } from 'fxa-admin-server/src/graphql';
+
+interface RateLimitSearchState {
+  ip: string;
+  email: string;
+  uid: string;
+}
+
+type RateLimitQuery = Partial<RateLimitSearchState>;
+
+export const PageRateLimiting = () => {
+  const { register, handleSubmit, reset, getValues } =
+    useForm<RateLimitSearchState>({
+      defaultValues: {
+        ip: '',
+        email: '',
+        uid: '',
+      },
+    });
+  const [showResults, setShowResults] = useState<boolean>(false);
+
+  const [getRateLimits, rateLimitsResult] = useLazyQuery(GET_RATE_LIMITS);
+  const [unblockMutation] = useMutation<{ unblock: number }>(UNBLOCK);
+  const [unbanMutation] = useMutation<{ unban: number }>(UNBAN);
+
+  const onSubmit = (data: RateLimitSearchState) => {
+    const { ip, email, uid } = data;
+    if (!ip.trim() && !email.trim() && !uid.trim()) {
+      window.alert('Please enter at least one of: IP address, email, or UID');
+      return;
+    }
+
+    const variables: RateLimitQuery = {};
+    if (ip.trim()) variables.ip = ip.trim();
+    if (email.trim()) variables.email = email.trim();
+    if (uid.trim()) variables.uid = uid.trim();
+
+    getRateLimits({ variables });
+    setShowResults(true);
+
+    // keep form values after submit
+    reset(data);
+  };
+
+  const handleUnblock = async () => {
+    const { ip, email, uid } = getValues();
+    if (!ip.trim() && !email.trim() && !uid.trim()) {
```

Comment:
Seems liek a lot of `.trim()` going on. Maybe just do this upfront? Then you know the values are good going forward. e.g.

```
const values = getValues();
const [ip, email, uid] = [values.ip.trim(), values.email.trim(), values.uid.trim()];
```